### PR TITLE
Jwt headers refactor (share with GeoNetwork)

### PR DESCRIPTION
**Pull request by davidblasby**
In order to facilitate compatibility (and less code) - I've broken the JWT Headers module into two modules;

`jwt-headers-util` - shared code that does the actual "work" of the headers extraction and JWT validation, etc...
`jwt-headers-gs` - geoserver specific code (i.e. filter and wicket)

The goal is for `jwt-headers-util` to be shared between Geoserver and Geonetwork.

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [x] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [ ] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [ ] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Comment by davidblasby** [comment]
I'll resolve conflicts - they are minor.

File: src/community/jwt-headers/jwt-headers-util/pom.xml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    <version>2.26-SNAPSHOT</version>
+  </parent>
+
+  <groupId>org.geoserver.community.jwt-headers</groupId>
+  <artifactId>jwt-headers-util</artifactId>
+  <packaging>jar</packaging>
+  <name>JWT Headers Shared Library</name>
+
+  <dependencies>
+    <!--    <dependency>-->
```
Comment (aaime) [inline src/community/jwt-headers/jwt-headers-util/pom.xml:24]: Please avoid commented-out code/configuration sections. Instead, have a comment here that no GeoServer dependency should be added (yes it's also in the README, I would not assume others to read it though, as there is no common practice to use READMEs to start with, in the GeoServer modules). It's a case where repeating oneself does not hurt.

File: src/community/jwt-headers/jwt-headers-gs/pom.xml [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.geoserver.community</groupId>
+    <artifactId>jwt-headers</artifactId>
+    <version>2.26-SNAPSHOT</version>
+  </parent>
+
+  <groupId>org.geoserver.community.jwt-headers</groupId>
+  <artifactId>jwt-headers-gs</artifactId>
```
Comment (mprins) [inline src/community/jwt-headers/jwt-headers-gs/pom.xml:19]: nitpicking here, GeoServer modules commonly have artifact-ids starting with `gs-` making it easier to recognise the jars in eg. an extracted webapp lib directory

File: src/community/jwt-headers/jwt-headers-gs/src/main/resources/applicationContext.xml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+          http://www.springframework.org/schema/beans/spring-beans-3.0.xsd
+          http://www.springframework.org/schema/security
+          http://www.springframework.org/schema/security/spring-security-3.0.4.xsd">
+
+	<bean class="org.geoserver.platform.ModuleStatusImpl">
+		<constructor-arg index="0" value="jwt-headers" />
+		<constructor-arg index="1" value="GeoServer JWT Headers Security" />
+	</bean>
+
+	<!-- enables Keycloak as an auth-provider -->
```
Comment (mprins) [inline src/community/jwt-headers/jwt-headers-gs/src/main/resources/applicationContext.xml:21]: should this be talking about keycloak?

File: src/community/jwt-headers/jwt-headers-gs/src/main/resources/applicationContext.xml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+
+	<bean id="jwtHeadersAuthPanelInfo" class="org.geoserver.security.jwtheaders.web.JwtHeadersAuthFilterPanelInfo">
+		<property name="id" value="security.jwtheader.headerAuthFilter" />
+		<property name="shortTitleKey" value="JwtHeadersAuthFilterPanel.short"/>
+		<property name="titleKey" value="JwtHeadersAuthFilterPanel.title"/>
+		<property name="descriptionKey" value="JwtHeadersAuthFilterPanel.description"/>
+	</bean>
+
+<!--	&lt;!&ndash; GeoServer UI for Keycloak auth config &ndash;&gt;-->
```
Comment (mprins) [inline src/community/jwt-headers/jwt-headers-gs/src/main/resources/applicationContext.xml:33]: can this comment block be removed?

File: src/community/jwt-headers/jwt-headers-util/pom.xml (hunk 1 above)
Comment (davidblasby) [reply src/community/jwt-headers/jwt-headers-util/pom.xml:24]: removed comments + added comment.

File: src/community/jwt-headers/jwt-headers-gs/pom.xml (hunk 2 above)
Comment (davidblasby) [reply src/community/jwt-headers/jwt-headers-gs/pom.xml:19]: changed to `gs-jwt-headers`

File: src/community/jwt-headers/jwt-headers-gs/src/main/resources/applicationContext.xml (hunk 3 above)
Comment (davidblasby) [reply src/community/jwt-headers/jwt-headers-gs/src/main/resources/applicationContext.xml:21]: I copied it from the keycloak module - changed!

File: src/community/jwt-headers/jwt-headers-gs/src/main/resources/applicationContext.xml (hunk 4 above)
Comment (davidblasby) [reply src/community/jwt-headers/jwt-headers-gs/src/main/resources/applicationContext.xml:33]: removed

**Comment by davidblasby** [comment]
Thanks for review @aaime and @mprins - it was helpful!

**Comment by davidblasby** [comment]
I will merge this tomorrow (thursday) if there's no more feedback.

**Comment by davidblasby** [comment]
Minor change to the pom's for `nimbus-jose-jwt` dependency.  This is so that downstream projects (GeoNetwork) can choose their own version a little easier (GS will use v9.37.3).

**Comment by davidblasby** [comment]
Added support for creating the release .zip.

cf. https://github.com/geoserver/geoserver/pull/7440#issuecomment-2027337759

**Pull request merged**